main/test/ca/uwaterloo/flix/TestVerifiers.scala

```diff
@@ -0,0 +1,47 @@
+package ca.uwaterloo.flix
+
+import ca.uwaterloo.flix.api.{Flix, FlixEvent, FlixListener}
+import ca.uwaterloo.flix.language.ast.shared.SecurityContext
+import ca.uwaterloo.flix.util.Options
+import ca.uwaterloo.flix.verifier.{EffectVerifier, TypeVerifier}
+import org.scalatest.funsuite.AnyFunSuite
+
+class TestVerifiers extends AnyFunSuite with TestUtils {
+
+  test("RunAllVerifiers") {
+    implicit val flix: Flix = new Flix()
+
+    flix.setOptions(Options.TestWithLibAll.copy())
+    flix.addListener(new FlixListener {
+      override def notify(e: FlixEvent): Unit = e match {
+        case FlixEvent.AfterTyper(root) =>
+          EffectVerifier.verify(root)
+        case FlixEvent.AfterTailPos(root) =>
+          TypeVerifier.verify(root)
+        case _ => ()
+      }
+    })
+    flix.addSourceCode("<test>", exampleProgram)(SecurityContext.AllPermissions)
+
+    val res = flix.compile()
+    expectSuccess(res)
+
+  }
+
+  /** "Complicated" Program to avoid tree shaking the whole AST away. */
+  private def exampleProgram: String =
```

Comment:
Why this specific program?

Reply:
It uses datalog which in turn uses a lot of stuff. It's just to avoid ThreeShaker2 to delete everything which would make backend verifiers not check anything